x509-cert/src/name.rs

```diff
@@ -2,17 +2,181 @@
 
 use crate::attr::AttributeTypeAndValue;
 use alloc::vec::Vec;
+use const_oid::{
+    db::{rfc3280, rfc4519},
+    ObjectIdentifier,
+};
 use core::{fmt, str::FromStr};
-use der::{asn1::SetOfVec, Encode};
+use der::{
+    asn1::{Any, PrintableStringRef, SetOfVec},
+    Encode,
+};
 
 /// X.501 Name as defined in [RFC 5280 Section 4.1.2.4]. X.501 Name is used to represent distinguished names.
 ///
 /// ```text
 /// Name ::= CHOICE { rdnSequence  RDNSequence }
 /// ```
 ///
+/// # Example
+///
+/// ```
+/// use std::str::FromStr;
+/// use x509_cert::name::Name;
+///
+/// let subject = Name::from_str("CN=example.com").expect("correctly formatted subject");
```

Comment:
This one example is insufficient to instruct users on how to produce strings that can be used to construct `Name`s.

Please add a section above this describing construction in terms of the "RFC 4514 string encoding", copying in the relevant attribute table and describing the concatenation syntax. Alternatively, add this content to the documentation of the `FromStr` impl, and link to it above here.

Reply:
I'm not sure which table you're referring to exactly.

Reply:
The one at the bottom of Section 3 of the RFC, that lists the stringified name of each supported attribute (e.g `CN`).

Reply:
Looks like the table has been copied and there are now many examples